### ISSUE TITLE
Continue pre-train for domain adaptation?

### ISSUE THREAD

**Issue by wj210** [facebookresearch/llama#816]
Hi, would like to ask if anyone has an opinion on this.

Would it be better to continue the pre-training or to do instruct-finetuning for domain adaptation?
The reason why I am leaning towards pre-training is due to the lack of supervised data and do not want to leverage LLMS like gpt4 to generate them (since the quality is limited by the domain knowledge).

However, if I were to continue the pre-training, I should use the base model rather than the chat-model? But I am wondering if I were to do that, since the base-model is not trained for safety (RLHF), it would possibly impact the quality of outputs.

**Comment by macqueen09** [comment]
In  my opinion , pre-train vs finetune is not so different between them.  because they are both train with a certern learning rate in your data (when you update all param).

base model or chat-model? It's depend your promblem , if your promblem is Generic tasks , you should use base model , otherwise    chat model. 

ONLY you update a part of param in model , fine tune has some different with pretrain. (freeze some layer , or LoRA  something else.)

**Comment by melanierk** [comment]
Thanks @macqueen09 for a good answer. I'll also add that, yes, continuing to train on any model can affect the safety of your end model, but especially the "base"/pretrained model since it has more limited safety mitigations. You should always evaluate your model for quality /safety after any kind of tuning or continuous pretraining, as explained in our responsible release guide: https://ai.meta.com/static-resource/responsible-use-guide/. We're hoping to release additional safety eval tools and libraries in the future to help with this kind of thing.
